### PR TITLE
[SPARK-44181][BUILD][3.4] Upgrade `Apache Kafka` to 3.4.1

### PR DESCRIPTION
**Pull request by bjornjorgensen**
### What changes were proposed in this pull request?
Upgrade Apache Kafka from 3.3.2 to 3.4.1 for branch 3.4

### Why are the changes needed?

[CVE-2023-25194](https://www.cve.org/CVERecord?id=CVE-2023-25194)

[Release Notes for 3.4.1](https://downloads.apache.org/kafka/3.4.1/RELEASE_NOTES.html)
Kafka 3.4.1 fixes 58 issues since the 3.4.0 release.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by bjornjorgensen** [comment]
@dongjoon-hyun

File: connector/kafka-0-10/src/test/scala/org/apache/spark/streaming/kafka010/KafkaRDDSuite.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ import scala.collection.JavaConverters._
 import scala.concurrent.duration._
 import scala.util.Random
 
-import kafka.log.{CleanerConfig, LogCleaner, LogConfig, UnifiedLog}
+import kafka.log.{CleanerConfig, LogCleaner, LogConfig, ProducerStateManagerConfig, UnifiedLog}
```
Comment (dongjoon-hyun) [inline connector/kafka-0-10/src/test/scala/org/apache/spark/streaming/kafka010/KafkaRDDSuite.scala:27]: Oh.. Sorry, I was confused.

File: pom.xml [hunk 2]
```diff
@@ -137,7 +137,7 @@
     <!-- Version used for internal directory structure -->
     <hive.version.short>2.3</hive.version.short>
     <!-- note that this should be compatible with Kafka brokers version 0.10 and up -->
-    <kafka.version>3.3.2</kafka.version>
+    <kafka.version>3.4.1</kafka.version>
```
Comment (dongjoon-hyun) [inline pom.xml:140]: My bad. I thought you suggested to upgrade from 3.4.0 to 3.4.1. 
We don't upgrade dependency in release branches when there is a breaking code change.

**Comment by dongjoon-hyun** [comment]
Given the code status, I'm unable to approve this. Shall we close this PR please, @bjornjorgensen ?

**Comment by bjornjorgensen** [comment]
ok, I thought that spark 3.4 was on kafka version 3.4, but it is on 3.3.2.

Yes, ok I will close this on. 
Thank you.

**Comment by dongjoon-hyun** [comment]
Thank you so much!

**Pull request closed without merging**